### PR TITLE
Adds two (three) MGLs.

### PR DESCRIPTION
**Pull request by LanCartwright**
One for crew (not added to maps) is the 4W40 MGL, which is pump action.
Adds 6B40 MGL to nukeops
Adds new 40mm grenades.

Will do changelog later.
Will post icons later.
Everything is tested.
This weapon is very fun to use.

:cl: 
rscadd: Adds semi-automatic Multiple Grenade Launcher.
rscadd: Adds pump-action Multiple Grenade Launcher.
rscadd: Adds 40mm Buckshot grenade shell.
rscadd: Adds 40mm Rubbershot grenade shell.
rscadd: Adds 40mm EMP-FSD grenade sabot.
rscadd: Adds 40mm Incindiary grenade sabot.
rscadd: Adds fins trash item.
rscadd: Adds strong ion bolt.
rscadd: Adds 40mm grenades to Uplink.
rscadd: Adds semi-automatic Multiple Grenade Launcher Uplink.
/:cl:

image
6B40 MGL (left) and 4W40 MGL (right).
Left to right.
40mm HE, Rubber, Buck, EMP-FSD, Incindiary.

image
Destroyed machines caused by EMP-FSD sabots. EMP emitting fins are scattered across the floor.

image
Warden suppresses assistants using 40mm rubbershot.

image
Syndicate sweeps several crew using 40mm buckshot.

image
Syndicate disables his wingman, the HOS, HOP, and Captain with several grenades.

### CODE REVIEW

**Comment by Xhuis** [comment]
# ___BECAUSE WE NEED MORE GUNS AM I RIGHT___

**Comment by LanCartwright** [comment]
One was removed recently.

This is unique.

**Comment by Supermichael777** [comment]
grenade launchers are fairly unique. It doesn't fire bullets of some new random calliber

**Comment by Xhuis** [comment]
I still don't support more traditional, ballistic/explosive weapons. This is _over five hundred_ years in the future and it would probably make more sense to have more advanced weapons.

**Comment by lzimann** [comment]
gib m90 back

**Comment by LanCartwright** [comment]
what do you want me to do, add a mech that is super fast but has same health as player

Then call it a power suit

Old, effective weapons will never be completely replaced by SS13 weapons.

Also I'm not sure I should even trust players with 40mm rubber and 40mm buckshot shells. 

Anyway they're super expensive... And I forgot to adjust the price of the grenade launcher to suit that.

As I was saying, these are super expensive and should be only used in special circumstances.

File: code/game/objects/effects/decals/Cleanable/misc.dm [hunk 1]
```diff
@@ -7,6 +7,11 @@
 	icon = 'icons/obj/objects.dmi'
 	icon_state = "shards"
 
+/obj/effect/decal/cleanable/generic/fins
```
Comment (KazeEspada) [inline code/game/objects/effects/decals/Cleanable/misc.dm]: should be trash not a decal.

File: code/modules/projectiles/guns/projectile/shotgun.dm [hunk 2]
```diff
@@ -354,3 +354,29 @@
...
+/obj/item/weapon/gun/projectile/shotgun/automatic/mgl
+	name = "6B40 MGL"
+	desc = "A semi-automatic 40mm grenade launcher with six revolving chambers, automatic ejection systems, and slot reloading."
+	icon_state = "mgl"
+	item_state = "mgl"
+	origin_tech = "combat=7;materials=4;syndicate=8"
+	fire_sound = 'sound/weapons/grenadelaunch.ogg'
+	pin = /obj/item/device/firing_pin/implant/pindicate
+	mag_type = /obj/item/ammo_box/magazine/internal/grenadelauncher/mgl
+	w_class = 3
```
Comment (KazeEspada) [inline code/modules/projectiles/guns/projectile/shotgun.dm]: fits in backpack

File: code/modules/projectiles/projectile/special.dm [hunk 3]
```diff
@@ -20,6 +20,23 @@
 	empulse(target, 0, 0)
 	return 1
 
+/obj/item/projectile/ion/strong
+	damage = 35
+	damage_type = TOX
+	nodamage = 0
+	weaken = 2
```
Comment (KazeEspada) [inline code/modules/projectiles/projectile/special.dm]: Isn't it an ion round, why does it do 35 tox?

File: code/game/objects/effects/decals/Cleanable/misc.dm (hunk 1 above)
Comment (LanCartwright) [inline code/game/objects/effects/decals/Cleanable/misc.dm]: Wilco. I'll change that to trash.

File: code/modules/projectiles/projectile/special.dm (hunk 3 above)
Comment (LanCartwright) [inline code/modules/projectiles/projectile/special.dm]: I think I was trying to get something going to how an EMP actually works. In hindsight I should probably put it back to brute damage instead.

File: code/modules/projectiles/guns/projectile/shotgun.dm (hunk 2 above)
Comment (LanCartwright) [inline code/modules/projectiles/guns/projectile/shotgun.dm]: yup.

the crew version doesn't fit in backpack though. That's on purpose.

**Comment by Dorsisdwarf** [comment]
I'm fairly sure this PR will be flat-out denied if the guns names are random jumbles of letters and numbers incomprehensible to someone who doesn't obsess over guns.

**Comment by Nienhaus** [comment]
ofsdeqb
I got you on the picture.

**Comment by LanCartwright** [comment]
@Dorsisdwarf 
I question the other guns if that's something that is taken seriously.
L6 SAW - L6? What is this. SAW? Squad Assault Weapon. Okay fair enough.
C20r - C20? 20 rounds. Whats the C for? r? Repeating? This makes no sense.
M90gl (now removed) - M90? It looks like a P90. Why an M?. gl - Grenade launcher.

There isn't really much reason to get confused on this. 6 or 4 is the capacity, then a random letter, then 40 for the 40mm grenades, mgl for Multiple Grenade Launcher.

**Comment by duncathan** [comment]
Changelog please

**Comment by Thunder12345** [comment]
Why would there be buckshot in a grenade launcher. Replace the silly 40mm shotgun shells with more sensible grenade payloads, like flashbang/teargas rounds.

**Comment by LanCartwright** [comment]
Trust me loading shotgun balls into a 40mm grenade has happened and it is extremely effective.

I'll see what I can do with flashbang and teargas.

**Comment by Iamgoofball** [comment]
need the S.M.K.E420 to go alongside the FK-69

**Comment by Dorsisdwarf** [comment]
-[ ] Fins need to get some offset like bullet casings do.

-[ ] Summon a spriter, I'll mail you the goat blood.

**Comment by LanCartwright** [comment]
Don't like the EMP shell?

Also I need to do something about the new 40mm shells since they're falling into a neat like rather than being scattered across the floor.

I'll get onto that tonight, plus a few other grenades (flashbang, tear gas, fungal tb, pipebomb, smoke) on impact.

File: code/modules/uplink/uplink_item.dm [hunk 4]
```diff
@@ -184,6 +184,14 @@ var/list/uplink_items = list() // Global list so we only initialize this once.
 	surplus = 25
 	include_modes = list(/datum/game_mode/nuclear)
 
+/datum/uplink_item/dangerous/mgl
+	name = "6B40 MGL"
+	desc = "New Russian semi-automatic grenade launcher. Has 6 chambers 40mm grenades."
```
Comment (KorPhaeron) [inline code/modules/uplink/uplink_item.dm]: Second half of this sentence is wrong

File: code/modules/uplink/uplink_item.dm [hunk 5]
```diff
@@ -184,6 +184,14 @@ var/list/uplink_items = list() // Global list so we only initialize this once.
...
+	item = /obj/item/weapon/gun/projectile/shotgun/automatic/mgl
+	cost = 25
```
Comment (KorPhaeron) [inline code/modules/uplink/uplink_item.dm]: Seems kinda cheap for a one hit kill weapon

Comment (LanCartwright) [inline code/modules/uplink/uplink_item.dm]: You're right. I think this might be the only gun that's going to break the 30 TC cost next round of commits.

**Pull request closed without merging**